src/package-denier/package-lock-file-util.ts

```diff
@@ -0,0 +1,105 @@
+// package-denier/package-lock-file-util.ts
+
+import path from 'node:path';
+import { readFile } from 'node:fs/promises';
+import type { Rule } from './rules';
+
+export interface Descriptor {
+  version: string;
+  resolved: string;
+  integrity: string;
+  link: boolean;
+  dev: boolean;
+  optional: boolean;
+  devOptional: boolean;
+  inBundle: boolean;
+  hasInstallScript: boolean;
+  hasShrinkwrap: boolean;
+  bin: string;
+  license: string;
+  engines: object;
+  dependencies: object;
+  devDependencies: object;
+  optionalDependencies: object;
+}
+
+export interface PackageLock {
+  name: string;
+  version: string;
+  lockfileVersion: number;
+  packages: { [key: string]: Descriptor };
+}
+
+export async function getPackageLock(rootProjectDirectory: string): Promise<PackageLock> {
+  const filePath = path.join(rootProjectDirectory, 'package-lock.json');
+  return JSON.parse(await readFile(filePath, 'utf8')) as PackageLock;
+}
+
+const prefix = 'node_modules/';
+
+export function getPackageNameFromKey(key: string): string {
+  return key.slice(key.lastIndexOf(prefix) + prefix.length);
+}
+
+function compareSemverVersions(version1: string, version2: string): number {
```

Comment:
would it make sense to use the npm semver package here? https://github.com/npm/node-semver - as its made by the NPM people its probably going to get updates for any additional operators they add? It includes a comparison operator